client/product-details/index.js

```diff
@@ -1,6 +1,40 @@
+/* global jQuery */
+// global wcpayStripeSiteMessaging
+
 /**
  * Internal dependencies
  */
-import bnplSiteMessaging from './bnpl-site-messaging';
+import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
+
+const bnplPaymentMessageElement = initializeBnplSiteMessaging();
+
+jQuery( function ( $ ) {
+	$( '.quantity input' ).on( 'change', function ( event ) {
+		const newQuantity = event.target.value;
+		const price = window.wcpayStripeSiteMessaging.price;
+		bnplPaymentMessageElement.update( {
+			amount: parseInt( price, 10 ) * newQuantity,
+		} );
+	} );
+
+	// Handle BNPL payment message changes for product variation.
+	const quantity = $( '.quantity input' ).val();
+	const { variationsPriceList } = window.wcpayStripeSiteMessaging;
+
+	$( '.single_variation_wrap' ).on( 'show_variation', function (
+		event,
+		variation
+	) {
+		const variationPrice = variationsPriceList[ variation.variation_id ];
+		window.wcpayStripeSiteMessaging.price = variationPrice;
```

Comment:
That's a bit hacky. Are there clearer alternatives here?

Reply:
I did feel a twitch in my left eye while writing this :trollface: 

To fix this, I have created a variable - `selectedVariationPrice`  within the script to record the prevailing variation price, instead of pushing it to the global object.  Let me know if this approach works. e451407